src/services/form-errors-message.service.ts

```diff
@@ -50,17 +50,25 @@ export class NgxFormErrorsMessageService {
 	 * the error message matching that group and the validation error. If no validation error matches both, then it returns the one matching the validation error.
 	 * Otherwise it returns undefined.
 	 * @param error - The validation error (Angular validator name)
+	 * @param formControlName - The FormControl name
 	 * @param group - The model group to find a match for (if any)
 	 */
-	public getErrorMessage(error: string, group?: string): string | undefined {
-		let errorKey: string = error;
+	public getErrorMessage(error: string, formControlName: string, group?: string): string | undefined {
```

Comment:
Mmm I don't think that `formControlName` should be really mandatory. What about the use cases where the developer calls this method somewhere in the app but not really related to any form control in particular? He should still be able to do so and get the generic error message for a certain validation right?

Reply:
If this is changed I would suggest changing the second param to a config and making it optional.

Reply:
A possible refactor of this function in its current state.
```
	public getErrorMessage(error: string, formControlName?: string, group?: string): string | undefined {
		return (
			(group && formControlName && this.errorMessages[`${group}.${formControlName}.${error}`]) ||
			(formControlName && this.errorMessages[`${formControlName}.${error}`]) ||
			(group && this.errorMessages[`${group}.${error}`]) ||
			this.errorMessages[error]
		);
	}
```

But i think it would actually make more sense for the user to be able to give the full path of the error in one go. Maybe something like this.
```
	public getErrorMessage(path: string | { error: string; formControlName?: string; group?: string }): string | undefined {
		path = typeof path === "string" ? path : [path.group, path.formControlName, path.error].filter(Boolean).join(".");
		return this.errorMessages[path];
	}
```

Reply:
After discussing the functionality of this `getErrorMessage` method with @cnomes, it seems that the name doesn't really describe the added value of this method, which is to "try to get the most suitable message". So we should rename this method to `findErrorMessage` so it is clear that the method will do its best to find the message that fulfills all (best case scenario) or some of the given parameters (acceptable case scenario). It will return `undefined` if no message with any of those params was found (worst case scenario).